File: types/notification.go
package types

type NotificationListReq struct {
	BasePage
	UnreadOnly bool `json:"unread_only" form:"unread_only"`
}

type NotificationMarkReadReq struct {
	IDs []uint `json:"ids" form:"ids"`
}

type NotificationUnreadResp struct {
	UnreadCount int64 `json:"unread_count"`
}

type NotificationResp struct {
	ID            uint   `json:"id"`
	RecipientType string `json:"recipient_type"`
	RecipientID   uint   `json:"recipient_id"`
	Scene         string `json:"scene"`
	Title         string `json:"title"`
	Content       string `json:"content"`
	Payload       string `json:"payload"`
	Read          bool   `json:"read"`
	CreatedAt     int64  `json:"created_at"`
}

File: web-admin/src/api/notification.ts
import request from "@/utils/request";

export interface NotificationItem {
  id: number;
  recipient_type: string;
  recipient_id: number;
  scene: string;
  title: string;
  content: string;
  payload: string;
  read: boolean;
  created_at: number;
}

export const getAdminNotificationList = (params: {
  page_num: number;
  page_size: number;
  unread_only?: boolean;
}) => request.get("/admin/notifications/list", { params });

export const getAdminNotificationUnreadCount = () =>
  request.get("/admin/notifications/unread_count", { silentError: true });

export const markAdminNotificationRead = (data: { ids: number[] }) =>
  request.post("/admin/notifications/read", data);

export const markAllAdminNotificationRead = () =>
  request.post("/admin/notifications/read_all");
